tests/data_generation/helper/analytics.py

```
@@ -0,0 +1,183 @@
+        """Print statistics about trigger code presence before simulation."""
+        total_subjects = df["subject_id"].nunique()
+        subject_codes = df.groupby("subject_id")["code"].apply(set)
+
+        print(f"\nTotal subjects: {total_subjects}")
+        print("\nTrigger code presence before simulation:")
+
+        # Get all unique codes and compute their presence once
+        all_codes = config.get_all_trigger_codes()
+        code_stats = {}
+        for code in all_codes:
+            count = subject_codes.apply(lambda codes: code in codes).sum()
```

Ruff (0.11.9)

Function definition does not bind loop variable `code`

(B023)